internal/rpc/jsonrpc/methods.go

```diff
@@ -2371,52 +2375,61 @@ func (s *Server) purchaseTicket(ctx context.Context, icmd interface{}) (interfac
 		}
 	}
 
-	// Set pool address if specified.
-	var poolAddr dcrutil.Address
-	var poolFee float64
-	if cmd.PoolAddress != nil {
-		if *cmd.PoolAddress != "" {
-			addr, err := decodeAddress(*cmd.PoolAddress, w.ChainParams())
-			if err != nil {
-				return nil, err
-			}
-			poolAddr = addr
-
-			// Attempt to get the amount to send to
-			// the pool after.
-			if cmd.PoolFees == nil {
-				return nil, rpcErrorf(dcrjson.ErrRPCInvalidParameter, "pool address set without pool fee")
-			}
-			poolFee = *cmd.PoolFees
-			if !txrules.ValidPoolFeeRate(poolFee) {
-				return nil, rpcErrorf(dcrjson.ErrRPCInvalidParameter, "pool fee percentage %v", poolFee)
-			}
-		}
-	}
-
```

Comment:
Removal of this seems to have broken creation of tickets with VSP voting (trying to call the `purchaseticket` json rpc method with a pooladdress returns a ticket without it).